EsBankAccount.Tests/Dsl/DeciderSpecs.fs

```diff
@@ -1,12 +1,12 @@
 namespace EsBankAccount.Tests.Domain
 
-open Swensen.Unquote
+open Diffract
 
 type DeciderSpecState<'State, 'Outcome> =
     { State: 'State
       Outcome: 'Outcome }
 
-type DeciderSpecList<'State, 'Command, 'Event>
+type DeciderSpecList<'State, 'Command, 'Event when 'Event: equality>
```

Comment:
That's a lot to ask of all events, don't you think?

🤔 I reckon this might be a mis-step in the long term as IME you want to be able to make more complex assertions beyond simple equality (you might want to have a range or rule about some field, though I guess that's straying into what you tend to end up doing in FsCheck more than in actual Favts like you have), but it really does depend I guess (and I have not personally tried Diffract and/or considered how one might combine Diffract and Unquote)

Reply:
(I guess my question is, have yo tried to port any messy real life prod tests to this scheme?)

Reply:
If we can get to a set of common aggregates with tests, I guess one can validate this DeciderSpec mechanism by applying it to those cases - hopefully we'll meet in a useful place (I've never imagined a world without unquote!)

Reply:
The DSL could be just a helper to have a nice BDD style. We can have different test styles in the same test file actually.

For event models, what kind of types couldn't have equality? (I can see for NoComparison but not for NoEquality)

Reply:
Yes only a structural diff is not really relevant, especially with custom equality and comparison. That's why I did this:
```fsharp
    if not (expected = spec.Outcome) then
        Diffract.Assert (expected, spec.Outcome)
```

I think I should add multiple overload for `Then`

- `'Event list`
- `'Error`

and also custom validations

- `'Event list -> unit | bool`
- `'Error -> unit | bool`

Reply:
Possibly - I wonder can the constraint move down to a specific method ?

I personally don't believe in a prescribed result type (i.e. for me a decider should return either `'event list` or `'result * 'event list`, and then the `Transact` layer yields the result (if the outcome got persisted))

That means that I want to be able to test Transact invocations with and without a result

And I dont want the presence of events to signify that its success

But I dont even have the imagination to try to work out a DSL so please don't let my kvetching guide you in this - the bottom line on this is the more thorny code you can make this cover, the more general and useful it will be, by definition!

Reply:
You're right🙏 I updated the readme so as not to sounds like the DSL would be mandatory and also I added issue #7